torch/_dynamo/variables/tensor.py

```diff
@@ -681,36 +684,59 @@ class TensorWithTFOverrideVariable(VariableTracker):
     def create(
         tx,
         tensor_variable,
-        orig_tensor_variable_source,
+        tensor_variable_source,
         torch_function_fn,
         subclass_type,
         **kwargs,
     ):
         var = TensorWithTFOverrideVariable(
             tensor_variable,
-            orig_tensor_variable_source,
+            tensor_variable_source,
             torch_function_fn,
             subclass_type,
             **kwargs,
         )
         # stash the subclass type to rewrap an output tensor if needed
-        tx.output.install_global(var.global_class_name(), subclass_type)
+        if var.global_class_name() not in tx.output.global_scope:
+            tx.output.install_global(var.global_class_name(), subclass_type)
+
         return var
 
     def __init__(
         self,
         tensor_variable,
-        orig_tensor_variable_source,
-        subclass_torch_function__func,
+        tensor_variable_source,
+        torch_function_fn,
         subclass_type,
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.tensor_variable = tensor_variable
-        self.orig_tensor_variable_source = orig_tensor_variable_source
-        self.subclass_torch_function__func = subclass_torch_function__func
+        self.tensor_variable_source = tensor_variable_source
+        self.torch_function_fn = torch_function_fn
         self.subclass_type = subclass_type
 
+    def as_proxy(self):
+        return self.tensor_variable.as_proxy()
+
+    def python_type(self):
+        return self.subclass_type
+
+    def var_getattr(self, tx, name: str) -> VariableTracker:
+        return self.tensor_variable.var_getattr(tx, name)
```

Comment:
This doesn't look right, the user could have easily defined attributes on the subclass (indeed, this is typically the raison d'etre of tensor subclassing), and there is no reason the vanilla TensorVariable should know how to deal with it!

Reply:
Yeah I think I did this to get around a very specific error, I'll handle it correctly instead.